hicommonwealth/commonwealth
# Update topic schema + add migration for weighted topics

## Link to Issue
Closes: #9132 

## Description of Changes
- Updates `Topics` schema to support weighted voting per topic
- Adds migration to update `Topics` table + apply 1-to-1 constraint for Contest to Topic
- Updates UI to restrict to 1 topic selection per contest

## Test Plan
- Create multiple contests, each assigned to multiple topics
- Ensure that the migration runs and each contest is afterwards only associated with 1 topic

## Deployment Plan
N/A

## Other Considerations
- There's data loss due to some contests going from multiple topics to a single topic. But the end-user impact will likely be minimal. At worst, any ongoing contests will only be active within a single topic.
- Technically, we should probably remove the `ContestTopics` table, but the surface area of that change would be quite large + the spec could change at any time.
